unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,149 @@
+import datetime as dt
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Generator, Optional
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+if TYPE_CHECKING:
+    from discord import Client as DiscordClient
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig] = Field(
+        default=DiscordAccessConfig, validate_default=True
+    )
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self) -> "DiscordClient":
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+
+class DiscordIndexerConfig(IndexerConfig):
+    channels: Optional[list[str]] = Field(
```

Comment:
Do we want to enforce at least one channel is provided? Otherwise this won't pull in anything that that might be odd to the end user.

Reply:
Good point. Then this list can't be optional. And for enforcing it being non-empty (at least one entry), the UI will have check it, because I don't see a way in backend code (Indexer or Downloader) to add a config check. Let me know if I'm missing sth :)